### PR TITLE
Column filters to most windows

### PR DESCRIPTION
**Pull request by Abyss777**
- Added columns filters to most windows
- Added more columns to `UserDevices` and `UserGroups`
- Changed openlayers version to 4.2.0

Implemented `groupId` formatter, unfortunately we need both stores `Groups` and `AllGroups`, make it as second parameter.
Or I could replace it with other key, like `groupIdAll` for example.

### CODE REVIEW

**Comment by tananaev** [comment]
Why do we need both for `groupId` formatter? `AllGroups` should have everything, right?

**Comment by Abyss777** [comment]
`AllGroups` request will fail for ordinary user https://github.com/tananaev/traccar/blob/master/src/org/traccar/api/resource/GroupResource.java#L48

**Comment by tananaev** [comment]
Maybe we can just change it to return list of user groups instead of failing. What do you think?

**Comment by Abyss777** [comment]
Then we should make the same for all other APIs for consistence ... Don't you think it looks like solving small inconvenience with changing some base things. ( I think API is some base ).

May be we can find the way on the client side only?

**Comment by tananaev** [comment]
How about we check if `AllGroups` is empty first. If it is, use `Groups`.

**Comment by Abyss777** [comment]
Good idea, or we can check if we are admin or manager.

**Comment by tananaev** [comment]
Yes, I think either option will work.

**Comment by Abyss777** [comment]
I have replaced commit with check `AllGroups` for empty.

**Comment by Abyss777** [comment]
Sorry, added small optimization.

File: web/app/AttributeFormatter.js [hunk 1]
```diff
@@ -62,6 +62,18 @@ Ext.define('Traccar.AttributeFormatter', {
         return Ext.getStore('Devices').getById(value).get('name');
     },
 
+    groupIdFormatter: function (value) {
+        var group;
+        if (value !== 0) {
+            if (Ext.getStore('AllGroups').getTotalCount() === 0) {
```
Comment (tananaev) [inline web/app/AttributeFormatter.js]: I think it's better to call `getStore` once. Do something like

```js
store = Ext.getStore('AllGroups');
if (store.getTotalCount() === 0) {
    store = Ext.getStore('Groups');
}
group = store.getById(value)
return group ? group.get('name') : value;

**Comment by tananaev** [comment]
Commented on the older commit, but I think it's still valid.

**Comment by Abyss777** [comment]
Done

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**